tracker/sm-website-update-visitor.js

```diff
@@ -1,147 +1,153 @@
+    return new Promise(function(resolve, reject) {
+     var xhr = new XMLHttpRequest();
+     var API_DOMAIN = "api.salemove.com"
+     var GET_CURRENT_VISIT_ENDPOINT = "https://" + API_DOMAIN + "/visitor";
+     xhr.open('GET', GET_CURRENT_VISIT_ENDPOINT);
+
+      requestHeaders = sm.salemoveApi.getRequestHeaders();
+        Object.keys(requestHeaders).forEach(function(key) {
+        xhr.setRequestHeader(key, requestHeaders[key]);
+     });
+
+      xhr.onload = function() {
+        if (xhr.status >= 200 && xhr.status < 300) {
+         parsedResponse = JSON.parse(xhr.responseText);
+         resolve(parsedResponse);
+        } else {
+         reject(xhr.statusText);
+        }
+      };
+      xhr.onerror = function() { reject(xhr.statusText); };
+
+     xhr.send();
+    });
+  }
+
   sm.getApi({version:'v1'}).then(
     function(api){
-      window.salemoveApi = api;
-      getHeaders();
-      getCurrentVisitor();
+      sm.salemoveApi = api;
 
-      // List of listeners for various events.
+        // List of listeners for various events.
 
-      // Type 1:
+        // Type 1:
         // Updates the current custom attributes
         // Then submits the object nested in custom attributes
       $('#preview').submit(function(){
         url = $('#cookieValue').val();
-        currentCustomAttributes["previewUrl"] = url
-        updateVisitorDetails("custom_attributes", currentCustomAttributes);
+        updateVisitorCustomAttributes("previewUrl", url);
       });
-      
+
       $('.resource, .home-persona-block').click(function(){
         personaType = this.dataset.persona;
-        currentCustomAttributes["persona"] = personaType
-        updateVisitorDetails("custom_attributes", currentCustomAttributes);
+        updateVisitorCustomAttributes("persona", personaType);
       });
 
       $('.home-video-colomn > .w-embed').click(function(){
-        currentCustomAttributes["videoWatched"] = "true"
-        updateVisitorDetails("custom_attributes", currentCustomAttributes);
+        updateVisitorCustomAttributes("videoWatched","yes")
       });
 
       $('#First-Name, #Last-Name').change(function(){
-        if ($(this).val() === "") {
-          return
-        } else {
-          name = $('#First-Name').val() + " " + $('#Last-Name').val()
-          updateVisitorDetails("name", name);
+        if ($(this).val().length > 0) {
+           name = $('#First-Name').val() + " " + $('#Last-Name').val()
+           updateVisitorInformation("name", name);
         };
       });
 
       $('#First-Name-3, #Last-Name-3').change(function(){
-        if ($(this).val() === "") {
-          return
-        } else {
+        if ($(this).val().length > 0 ) {
           name = $('#First-Name-3').val() + " " + $('#Last-Name-3').val()
-          updateVisitorDetails("name", name);
+          updateVisitorInformation("name", name);
         }
       });
 
       $('#First-Name-2, #Last-Name-2').change(function(){
-        if ($(this).val() === "") {
```

Comment:
You mean the nature of 
if true
  do nothing
else 
  do the code
and if so, is that the only difference between > 0 and "" in this case?